### PR TITLE
Launchpad: Enable the design-first flow to use the new task definition

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to https://github.com/Automattic/wp-calypso/issues/86647

## Proposed Changes
Enable the design-first flow to use the new task definition parser for testing purposes, since it is using already migrated tasks. 

'design_completed',
'setup_blog',
'domain_upsell',
'plan_completed',
'first_post_published',
'blog_launched',
				
All tasks related to this flow were already migrated, so this PR is just enabling the feature flag.

## Testing Instructions
* Start the flow ` design-first`  (setup/design-first)
* Follow the flow until arrive on the launchpad page (setup/start-writing/launchpad)
* Enable the feature flags via url
* Check if  there are messages on devtools to all tasks above
* Use the launchpad (Set values, Edit values, Check if the values are marked when completed)
* Finish the flow.

Requires the feature flags:
* "launchpad/new-task-definition-parser"
* "launchpad/new-task-definition-parser/design-first"

The flags can be enabled appending  the `?flags=` flag on the url (Please always check if the flags= param is still on the url when the system redirect back the user from some "no launchpad" page ).

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by daledupreez — COMMENTED** [review]
@gabrielcaires, there are a number of PRs around this switch-over, but I am missing some more general context around what you're doing and how everything ties together. I think it might be helpful to create a tracking issue that describes what you're doing, how you expect to roll out the changes, and what all the interconnected PRs are (as well as any dependencies between them). Even without the above, it would be good to add slightly more context to each PR to describe the broader project and what parts of the project each PR is addressing, as well as any PRs that we're depending on (like the PR to introduce some of the feature flags).

**Comment by gabrielcaires** [comment]
> @gabrielcaires, there are several PRs around this switch-over, but I am missing some more general context around what you're doing and how everything ties together. I think it might be helpful to create a tracking issue that describes 

@daledupreez 
Thanks for the feedback, I was feeling the same and you confirmed this perspection. I created this issue https://github.com/Automattic/wp-calypso/issues/86647 providing a bit more context.

I want to highlight I was not expecting this number of PRs but since we have a huge coupling on the original code, the dependent PRs are necessary.  The expectation is to have the new state be better to evolve.

File: config/development.json [hunk 1]
```diff
@@ -112,7 +112,8 @@
 		"launchpad/navigator": false,
 		"launchpad/new-task-definition-parser": true,
 		"launchpad/new-task-definition-parser/free": true,
-		"launchpad/new-task-definition-parser/start-writing": false,
+		"launchpad/new-task-definition-parser/start-writing": true,
```
Comment (gabrielcaires) [inline config/development.json:115]: Fixing a merging issue.
It is only for development.

**Review by markbiek — APPROVED** [review]
LGTM!

**Pull request merged**